### PR TITLE
Xds sender bug fixes

### PR DESCRIPTION
**Pull request by areklalo**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/shr/cdahandler/processor/util/AssignedEntityProcessorUtil.java [hunk 1]
```diff
@@ -238,7 +239,20 @@ else if(assignedEntity.getId() == null || assignedEntity.getId().isNull() || ass
 		// TODO: How to add this like the ECID/EPID identifiers in the current SHR
 		// ie. root becomes an attribute and the extension becomes the value
 		// Anyways, for now represent in the standard ITI guidance for II data type
-		String id = this.m_datatypeUtil.formatIdentifier(assignedEntity.getId().get(0));
+        String id = this.m_datatypeUtil.emptyIdString();
+        String epidRoot = this.m_configuration.getEpidRoot();
+        if (!epidRoot.equals("") && !epidRoot.isEmpty()) {
```
Comment (pgesek) [inline api/src/main/java/org/openmrs/module/shr/cdahandler/processor/util/AssignedEntityProcessorUtil.java]: isn't this the same check?

File: api/src/main/java/org/openmrs/module/shr/cdahandler/processor/util/LocationOrganizationProcessorUtil.java [hunk 2]
```diff
@@ -121,11 +121,19 @@ private Location createLocation(Organization organization, String id) throws Doc
 	 */
 	public Location getOrganizationByExternalId(String id) throws DocumentImportException
 	{
+		try {
+			Integer locationId = Integer.valueOf(id.substring(0, id.indexOf("^")));
+			Location locationById = Context.getLocationService().getLocation(locationId);
+			if (locationById != null)
+				return locationById;
+		} catch (StringIndexOutOfBoundsException ex) {
```
Comment (pgesek) [inline api/src/main/java/org/openmrs/module/shr/cdahandler/processor/util/LocationOrganizationProcessorUtil.java]: we should rather an if here, if (locationId == -1), then skip or something along those line

File: api/src/main/java/org/openmrs/module/shr/cdahandler/processor/util/AssignedEntityProcessorUtil.java (hunk 1 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/shr/cdahandler/processor/util/AssignedEntityProcessorUtil.java]: Corrected

File: api/src/main/java/org/openmrs/module/shr/cdahandler/processor/util/LocationOrganizationProcessorUtil.java (hunk 2 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/shr/cdahandler/processor/util/LocationOrganizationProcessorUtil.java]: Corrected

**Review by pgesek — APPROVED** [review]

**Pull request merged**